### ISSUE TITLE
System doesn't connected itself even after Mainnet spork is completed

### ISSUE THREAD

**Issue by rpatidar-isystango** [flow-hydraulics/flow-wallet-api#288]
After the 6th April 2022 Mainnet Spork completion. Even after restarting it was not able to connect to Chain and the service was continuously throwing `level=warning msg="Unable to connect to chain, pausing system" function=Listener.Start.goroutine package=chain_events`. From the local system, the chain was getting connected but not from my production server. Then I created a new DB and connect the service to the newly created DB and there it was connected. not sure what status was being managed in DB. Only one configuration I found in the **system_settings** table where it was managing **paused_since** value with some date and time. but even setting it to null it was automatically reset to a DateTime.

I have requested the wallet-api to create/setup some accounts but at that time mainnet was down(Sporked) so API failed and after failing it was continuously retied creating accounts until we went out of Flows

**Comment by sheerryy** [comment]
@arpit-systango @rpatidar-isystango Did you find any useful information to resolve this issue?

**Comment by rpatidar-isystango** [comment]
Hi Shehryar,

Thank you for writing back.

We are still waiting for that fix. I can see lots of people complaining
about the same issue with this library. Whenever Flow Mainnet goes
down/restart/fork every time the wallet gets stuck with the message "chain
is not connected" and does not connect itself back when the chain is up
again. And to fix that every time we set the block_height to 0 in DB and
restarts the service manually. It happens with Mainnet mostly but not with
Testnet. One more issue is if the wallet is disconnected still it receives
the Account creation request and puts that in JOB, and keeps calling the
Mainnet until the account gets out of the flow. It happened two times with
us.

Let us know if you need more information. Looking forward to hearing from
you.

Regards,
Rohit

**Comment by rpatidar-isystango** [comment]
> @arpit-systango @rpatidar-isystango Did you find any useful information to resolve this issue?

Nope, no response till... but we have found a manual way to resolve it... First, you need to stop the service then there is a block_height column in system_settings table where you can set the value to 0, and after that restart the service. It will reconnect the chain without a problem.

**Comment by sheerryy** [comment]
@rpatidar-isystango 🙌 Thanks man it worked. On a side note, my jobs are taking a lot of time from changing the state from `INIT` to `ACCEPTED`. any insight on this?

**Comment by rpatidar-isystango** [comment]
> @rpatidar-isystango raised_hands Thanks man it worked. On a side note, my jobs are taking a lot of time from changing the state from `INIT` to `ACCEPTED`. any insight on this?

Try Webhook and async calls, they are better than polling.

**Comment by sheerryy** [comment]
> > @rpatidar-isystango raised_hands Thanks man it worked. On a side note, my jobs are taking a lot of time from changing the state from `INIT` to `ACCEPTED`. any insight on this?
> 
> Try Webhook and async calls, they are better than polling.

We are already using async. How much time does it takes normally to complete an async transaction on your end?

**Comment by rpatidar-isystango** [comment]
> > > @rpatidar-isystango raised_hands Thanks man it worked. On a side note, my jobs are taking a lot of time from changing the state from `INIT` to `ACCEPTED`. any insight on this?
> > 
> > 
> > Try Webhook and async calls, they are better than polling.
> 
> We are already using async. How much time does it takes normally to complete an async transaction on your end?

It's based on the method. Account creation takes 10-15 sec along with the account setup and minting takes 30-40 sec.. but varies on Tesnet and Mainnet. I personally feel, the TestNet is faster than the MainNet
